src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1402,7 +1402,9 @@ private function registerSerializerConfiguration(array $config, ContainerBuilder
         }
 
         if ($config['max_depth_handler'] ?? false) {
-            $container->getDefinition('serializer.normalizer.object')->addMethodCall('setMaxDepthHandler', [new Reference($config['max_depth_handler'])]);
+            $arguments = $container->getDefinition('serializer.normalizer.object')->getArguments();
+            $context = ($arguments[6] ?? []) + ['max_depth_handler' => new Reference($config['max_depth_handler'])];
```

Comment:
why is this conditional? don't we always know if it's define or not because of `serializer.xml`?
also, where is argument `5` defined? (we cannot define arg 6 if 5 is not set afaik)

Reply:
the arguments 5 and 6 were not defined on `serializer.xml`. Now they are.
[https://github.com/symfony/symfony/pull/31117/files#diff-6cd5bb767c43a9c44442521122006d29R65](url)

With the new definition i can remove this conditional and use the `key` of the argument on the Config and the UnitTest
[https://github.com/symfony/symfony/pull/31117/files#d](url)
[https://github.com/symfony/symfony/pull/31117/files#diff-b0c828e732d3d31f0b2bfb2378bd1011R1070](url)